webapi/src/main/scala/org/knora/webapi/routing/v2/ResourcesRouteV2.scala

```diff
@@ -176,16 +177,43 @@ class ResourcesRouteV2(routeData: KnoraRouteData) extends KnoraRoute(routeData)
                         } yield requestMessage
 
                         RouteUtilV2.runRdfRouteWithFuture(
-                            requestMessageFuture,
-                            requestContext,
-                            settings,
-                            responderManager,
-                            log,
-                            ApiV2WithValueObjects
+                            requestMessageF = requestMessageFuture,
+                            requestContext = requestContext,
+                            settings = settings,
+                            responderManager = responderManager,
+                            log = log,
+                            responseSchema = ApiV2WithValueObjects
                         )
                     }
                 }
             }
+        } ~ path("v2" / "resources" / "history" / Segment) { resourceIriStr: IRI =>
+            get {
+                requestContext => {
+                    val resourceIri = stringFormatter.validateAndEscapeIri(resourceIriStr, throw BadRequestException(s"Invalid resource IRI: $resourceIriStr"))
+                    val params: Map[String, String] = requestContext.request.uri.query().toMap
+                    val startDate: Option[Instant] = params.get("startDate").map(dateStr => stringFormatter.xsdDateTimeStampToInstant(dateStr, throw BadRequestException(s"Invalid start date: $dateStr")))
+                    val endDate = params.get("endDate").map(dateStr => stringFormatter.xsdDateTimeStampToInstant(dateStr, throw BadRequestException(s"Invalid end date: $dateStr")))
+
+                    val requestMessageFuture: Future[ResourceVersionHistoryGetRequestV2] = for {
+                        requestingUser <- getUserADM(requestContext)
+                    } yield ResourceVersionHistoryGetRequestV2(
+                        resourceIri = resourceIri,
+                        startDate = startDate,
+                        endDate = endDate,
+                        requestingUser = requestingUser
+                    )
+
+                    RouteUtilV2.runRdfRouteWithFuture(
+                        requestMessageF = requestMessageFuture,
+                        requestContext = requestContext,
+                        settings = settings,
+                        responderManager = responderManager,
+                        log = log,
+                        responseSchema = ApiV2WithValueObjects
+                    )
+                }
+            }
         } ~ path("v2" / "resources" / Segments) { resIris: Seq[String] =>
```

Comment:
Here I would prefer:

```
/v2/resources/[resourceIRI]/version/[timestamp]
```

Reply:
and for multiple IRIs:

```
/v2/resources/version/[timestamp]?Id=iri1,iri2,iri3
```

Reply:
I think that the more information is in the URL path the better. I don't like URL parameters very much.

Reply:
I don't know whether changing this would break Knora-ui, BEOL, etc.

Reply:
Definitely. The comments are only meant as "food for thought". We can make an issue and do it sometime when @kilchenmann has more time.